ooniprobe/Engine/Engine.h

```diff
@@ -1,26 +1,42 @@
 #import <Foundation/Foundation.h>
-#import "ExperimentSettings.h"
-#import "MKExperimentTaskAdapter.h"
-#import "MKGeoIPLookupResultsAdapter.h"
-#import "MKGeoIPLookupTaskAdapter.h"
-#import "MKReporterResultsAdapter.h"
-#import "MKReporterTaskAdapter.h"
-#import "OONIProbeEngineTaskAdapter.h"
+#import "OONIMKTask.h"
+#import "OONIMKTaskConfig.h"
+#import "PEMKTask.h"
+#import "OONIContext.h"
+#import "OONISession.h"
+#import "OONISessionConfig.h"
+#import "PESession.h"
+#import "OONILogger.h"
+#import "LoggerComposed.h"
+#import "LoggeriOS.h"
+#import "LoggerNull.h"
+#import "LoggerArray.h"
 
 /**
 * Engine is a factory class for creating several kinds of tasks. We will use different
 * engines depending on the task that you wish to create.
 */
 @interface Engine : NSObject
 
-+ (NSString*) versionMK;
-
 + (NSString*) newUUID4;
 
-+ (id<ExperimentTask>) startExperimentTaskWithSettings:(id<ExperimentSettings>)settings error:(NSError **)error;
++ (id<OONIMKTask>) startExperimentTaskWithSettings:(id<OONIMKTaskConfig>)settings error:(NSError **)error;
+
++ (NSString*) resolveProbeCCWithSoftwareName:(NSString*)softwareName
+                             softwareVersion:(NSString*)softwareVersion
+                                     timeout:(long)timeout
+                                       error:(NSError **)error;
+
++ (PESession*) newSession:(OONISessionConfig*) config error:(NSError **)error;
```

Comment:
```suggestion
+ (id<OONISession>) newSession:(OONISessionConfig*) config error:(NSError **)error;
```

We probably want to avoid exposing `PESession`?